### PR TITLE
Syntax updates

### PR DESCRIPTION
**Pull request by fleiner**
Hi Bram,

here a pull request for the various changes that came up.

thanks, Claudio

### CODE REVIEW

**Comment by fleiner** [comment]
One of the commits is wrong, mentioning the wrong person

**Comment by chrisbra** [comment]
When you create the next PR, may I ask you to please check the following PRs and check which changes you'd like to have included (and thus which other PRs we can then close) please?

- [javascript.vim: modified number literals regex](https://github.com/vim/vim/pull/10520)
- [Add syntax support for Java switch expressions](https://github.com/vim/vim/pull/9124)
- [Keywords 'of' and 'yield' for Javascript](https://github.com/vim/vim/pull/7873)
- [Add new HTML 5 tags](https://github.com/vim/vim/pull/1263)


Thank you!

**Comment by brammool** [comment]
I had already included this before I noticed it was closed again.
One remark: It's not needed to add an exclamation mark after "function", I removed it.

File: runtime/syntax/m4.vim [hunk 1]
```diff
@@ -21,6 +21,7 @@ endif
 syn match  m4Variable contained "\$\d\+"
 syn match  m4Special  contained "$[@*#]"
 syn match  m4Comment  "\<\(m4_\)\=dnl\>.*" contains=SpellErrors
+syn match  m4Comment  "#.*" contains=SpellErrors
```
Comment (dkearns) [inline runtime/syntax/m4.vim:24]: SpellErrors is pretty old tech.  Should we update this to use the `@Spell` cluster?

**Pull request closed without merging**